src/MPDevice.cpp

```diff
@@ -6766,6 +6779,119 @@ bool MPDevice::checkImportedDataNodes(const MessageHandlerCb &cb, Common::DataAd
+                        auto* importedLastUsedChildNode = findNodeWithAddressWithGivenParentInList(importedLoginChildNodes, importedLoginNode, importedLoginNode->getLastChildNodeUsedAddr(), 0);
+                        if (importedLastUsedChildNode)
+                        {
+                            auto lastUsedChildName = importedLastUsedChildNode->getLogin();
+                            qDebug() << "Last used child name: " << lastUsedChildName;
+                            // Find the node with the same login name for the given service
+                            auto* lastUsedChild = findNodeWithLoginWithGivenParentInList(loginChildNodes, loginNode, lastUsedChildName);
+                            if (lastUsedChild && lastUsedChild->getAddress() != bleLoginNode->getLastChildNodeUsedAddr())
+                            {
+                                // Set the last child node used address to the correct login
+                                bleLoginNode->setLastChildNodeUsedAddr(lastUsedChild->getAddress());
+                            }
+                        }
+                    }
+                    else if (Common::NOT_SET_ADDR != bleLoginNode->getLastChildNodeUsedAddr())
+                    {
+                        bleLoginNode->setLastChildNodeUsedAddr(Common::NOT_SET_ADDR);
+                    }
+                    break;
+                }
+            }
+        }
+    }
+    return true;
+}
+
+bool MPDevice::checkImportedPointedToAddresses()
+{
+    if (!isBLE())
+    {
+        qCritical() << "Last child node used address is only available for BLE";
+        return false;
+    }
+
+    for (auto* loginNode : qAsConst(loginNodes))
+    {
+        // If service is not tagged for merge it will be removed, so do not need to check
+        if (loginNode->getMergeTagged())
+        {
+            const auto serviceName = loginNode->getService();
+            auto childAddr = loginNode->getStartChildAddress();
+            // Find the parent node for the same service in imported list
+            auto* importedLoginNode = findNodeWithNameInList(importedLoginNodes, serviceName, true);
+            // Iterating through all the parent's children
+            while (childAddr != Common::NOT_SET_ADDR)
+            {
+                auto* childNode = findNodeWithAddressInList(loginChildNodes, childAddr);
+                // If login is tagged for merge check the pointed to address for the node
+                if (childNode && childNode->getMergeTagged())
+                {
+                    auto* bleChildNode = static_cast<MPNodeBLE*>(childNode);
+                    // Find the child node for the same login name in imported list
+                    auto* importedChildNode = findNodeWithLoginWithGivenParentInList(importedLoginChildNodes, importedLoginNode, childNode->getLogin());
+                    if (importedChildNode)
```

Comment:
no else() needed here as we may do an additive merge.. though we could also check that merge condition